src/software/world/game_state.h

```diff
@@ -357,4 +345,23 @@ class GameState
      *         in a state of play where the ball needs to be placed
      */
     std::optional<Point> getBallPlacementPoint() const;
+
+    State getState() const;
+
+    void setState(State state);
+    void setRestartReason(RestartReason restart_reason);
+    void setOurRestart(bool our_restart);
+
+   private:
+    State state_;
+    RestartReason restart_reason_;
+    RefboxGameState game_state_;
+    std::optional<Ball> ball_state_;
+
+    // True if our team can kick the ball during a restart
+    bool our_restart_;
+
+    // The point at which the ball should be placed by robots before a restart. See
+    // Robocup SSL Rules 9.2.
```

Comment:
Is this referring to "The robot is at a distance from the halfway line that must not exceed 1 meter" from `9. Robot Substitution`? There isn't an explicit 9.2 (Source: https://robocup-ssl.github.io/ssl-rules/sslrules.html#_robot_substitution)

Reply:
I think it's referring to http://wiki.robocup.org/images/7/73/Small_Size_League_-_Rules_2018.pdf but it seems to not be up

Reply:
If possible, can you update the comment to reflect the intended rule?

Reply:
this is probably out of scope